### DISCUSSION TITLE
有两个问题

### DISCUSSION THREAD

**Discussion by shaoyangshujian** [Tokinx/Wing#66] [讨论]
1.私密和加密码的文章前台无法显示
2.不知道什么原因 右侧**更新日历** 不显示任何更新和评论的亮点。似乎这个功能一直失效好久。也许是个人原因，当登陆账号登陆后台后，会显示亮点，当退出后台过一段时间又不显示亮点了。我用的宝塔的配置环境。

**Comment by shaoyangshujian** [comment]
这个主题我也许会用一辈子。

**Reply by shaoyangshujian** [reply]
如果能解决更新日历的亮点不显示问题就完美了。

**Comment by Tokinx** [comment]
1、加密码的文章会在前台显示吧，这个我有特别适配的，可以看这里：https://inarave.com/category/the-cloud
2、私密文章只有在登陆时才显示
3、日历问题，登陆后正常退出登录后异常。
综合上面三个特征，你一定是开了某些缓存插件，请关闭它们，然后清空浏览器缓存在重新尝试

**Reply by shaoyangshujian** [reply]
模板设置选择：您的最新文章 会显示密码文章。 如果用使用一个静态页，静态页选择一个页面的首页模板就不会显示加密。
另外我完全没有开启缓存。
![屏幕截图 2022-11-04 154223](https://user-images.githubusercontent.com/111892369/199919387-fba77e57-cad5-4c0b-bfb4-7fa87a359198.png)

**Reply by shaoyangshujian** [reply]
【更新日历 不显示任何更新和评论的亮点】我知道是什么问题引起的了。只要登录WP后台随便编辑一篇文章，日历就会显示。一旦用WordPressAPP、OpenLiveWriter和南博APP的客户端新增文章后，日历的亮点就瞬间消失。所以我猜是不是用APP等博客客户端更新文章之后就会把某些暂存的数据删除掉？因为我一直用APP更新，怎么才能避免这种情况发生呢？

**Reply by Tokinx** [reply]
> 【更新日历 不显示任何更新和评论的亮点】我知道是什么问题引起的了。只要登录WP后台随便编辑一篇文章，日历就会显示。一旦用WordPressAPP、OpenLiveWriter和南博APP的客户端新增文章后，日历的亮点就瞬间消失。所以我猜是不是用APP等博客客户端更新文章之后就会把某些暂存的数据删除掉？因为我一直用APP更新，怎么才能避免这种情况发生呢？

可能是这样，数据是缓存在WP提供的瞬态缓存下的，可能第三方APP通过接口更新会清空瞬态缓存？这个我不确定，最近公司项目有点忙，而且我本地的环境挂掉了，暂时没办法更新主题，我下次更新看看加一下选择是否将数据缓存下来，或者看下有没有其它解决方案。

**Reply by shaoyangshujian** [reply]
> > 【更新日历 不显示任何更新和评论的亮点】我知道是什么问题引起的了。只要登录WP后台随便编辑一篇文章，日历就会显示。一旦用WordPressAPP、OpenLiveWriter和南博APP的客户端新增文章后，日历的亮点就瞬间消失。所以我猜是不是用APP等博客客户端更新文章之后就会把某些暂存的数据删除掉？因为我一直用APP更新，怎么才能避免这种情况发生呢？
> 
> 可能是这样，数据是缓存在WP提供的瞬态缓存下的，可能第三方APP通过接口更新会清空瞬态缓存？这个我不确定，最近公司项目有点忙，而且我本地的环境挂掉了，暂时没办法更新主题，我下次更新看看加一下选择是否将数据缓存下来，或者看下有没有其它解决方案。


--------------
经过如下测试
1、同一环境下安装新的WP没有任何问题，并且为了排除插件、文件的差异把不显示“日历”的网站程序文件完整包都覆盖在新安装的WP，启用和设置相同的插件。都是没任何问题。【结论：排除掉模板文件、服务器环境、程序文件】。
2、不显示“日历”的网站，删除掉数据库重新导入备份数据库。问题依旧。【结论：排除掉数据库版本或编码问题。】
3、我将不显示“日历”的旧网站数据库，导入到新安装的WP数据中。发现新安装的网站“日历”出现跟老网站一样的异常。即APP通过接口更新文章后，瞬态也没有问题。但就是前端不显示。只有登录后台打开文章列表后，前台才显示。当用APP更新又瞬间不显示了。【结论：确定是数据库某些表和内容带来的异常】

有此推断，出问题的是个例，而且只与数据库中的内容或者表有关。但是我不知道如何进一步排查和解决。如果你需要我可以发一份数据库备份到您邮箱里。
再次感谢！

**Comment by shaoyangshujian** [comment]
https://inarave.com/category/the-cloud   这个文章列表页貌似也没有显示加密文章，好像只有瀑布流的方块样式才会显示加密文章

**Reply by Tokinx** [reply]
> https://inarave.com/category/the-cloud 这个文章列表页貌似也没有显示加密文章，好像只有瀑布流的方块样式才会显示加密文章

大概了解了，你的意思是笔记模板页面没有显示加密、私密的文章对吗？等最近这段时间忙完我来更新这部分。最近有些忙，本地环境挂掉了，等忙完我重装一下本地环境。我把这个问题记录到更新计划里，具体更新时间不确定（可能要1个月后），暂且忍耐一下哈。
